osbuild/loop.py

```diff
@@ -126,6 +126,12 @@ def __init__(self, minor, dir_fd=None):
             if not dir_fd:
                 dir_fd = os.open("/dev", os.O_DIRECTORY)
                 stack.callback(lambda: os.close(dir_fd))
+            # If the loopdev didn't show up for some reason let's
+            # create it manually
+            try:
+                os.stat(self.devname, dir_fd=dir_fd)
+            except FileNotFoundError:
+                self.mknod(dir_fd)
```

Comment:
There's a bit of a TOCTOU race here, but that might be okay.  I'm more worried about putting a band-aid over a bigger issue.  What are the situations where we might be initialising a Loop for an fd that doesn't exist?

Reply:
When I was developing https://github.com/osbuild/osbuild/pull/1947 I hit a few situations where I had to ctrl-c out of running builds and I ended up in a situation where my VM (for some reason) wouldn't create the loop devices and would hit an error here in the `os.open()` call below.

A check and manual create here seemed to fix it.

It may have been an intermediate version of my code that didn't clean up properly or some kernel bug, but I figured this check couldn't hurt.

I can drop it if we'd like and we can add it back when/if people hit issues like the one I was hitting.

Reply:
> I hit a few situations where I had to ctrl-c out of running builds

Note one thing I did notice during development here is that if the Server side of this code hits an exception the client side doesn't ever get a message back and the code inside the stage just waits forever (hence the `ctrl-c` to get out).

Reply:
> Note one thing I did notice during development here is that if the Server side of this code hits an exception the client side doesn't ever get a message back and the code inside the stage just waits forever (hence the `ctrl-c` to get out).

That sounds like a bigger issue we should look into.



> I can drop it if we'd like and we can add it back when/if people hit issues like the one I was hitting.

Let's do that.